docs/articles/documentation/test-api/selecting-page-elements/selector-options.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: docs
+title: Selector Options
+permalink: /documentation/test-api/selecting-page-elements/selector-options.html
+---
+# Selector Options
+
+You can pass the following options to the [Selector constructor](selectors.md#creating-selectors) or the
+[t.select](selectors.md#one-time-selection) function.
+
+* [options.dependencies](#optionsdependencies)
+* [options.boundTestRun](#optionsboundtestrun)
+* [options.textFilter](#optionstextfilter)
+* [options.index](#optionsindex)
+* [options.timeout](#optionstimeout)
+* [options.visibilityCheck](#optionsvisibilitycheck)
+
+You can also overwrite the specified options at any time.
+
+* [Overwriting Options](#overwriting-options)
+
+## options.dependencies
+
+**Type**: Object
+
+Contains functions, variables or objects used by the selector internally.
+Properties of the `dependencies` object will be added to the selector's scope as variables.
+
+The following sample demonstrates a selector (`getGridCell`) that calls another selector (`getGridRow`) internally.
+The `getGridRow` selector is passed to `getGridCell` as a dependency.
+
+```js
+import { Selector } from 'testcafe';
+
+const getGridRow  = Selector(n => document.getElementsByClassName('grid-row')[n]);
+const getGridCell = Selector((m, n) => getGridRow(m).children[n], {
+     dependencies: { getGridRow }
+});
+```
+
+## options.boundTestRun
+
+**Type**: Object
+
+If you need to call a selector from a Node.js callback, assign the current
+[test controller](../test-code-structure.md#test-controller) to the `boundTestRun` option.
+
+For details, see [Calling Selectors from Node.js Callbacks](selectors.md#calling-selectors-from-nodejs-callbacks).
+
+## options.textFilter
+
+**Type**: String &#124; RegExp
+
+Text or a regular expression that matches text content of the node to be selected.
```

Comment:
Important part of functionality, as well as `index` - requires example and highlight in the main article